platform/xdebugger-impl/src/com/intellij/xdebugger/impl/XDebuggerActionsCollector.kt

```diff
@@ -0,0 +1,90 @@
+// Copyright 2000-2023 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+package com.intellij.xdebugger.impl
+
+import com.intellij.internal.statistic.collectors.fus.fileTypes.FileTypeUsagesCollector
+import com.intellij.internal.statistic.eventLog.EventLogGroup
+import com.intellij.internal.statistic.eventLog.events.EventFields
+import com.intellij.internal.statistic.eventLog.events.EventFields.Int
+import com.intellij.internal.statistic.eventLog.events.EventFields.IntList
+import com.intellij.internal.statistic.eventLog.events.EventFields.String
+import com.intellij.internal.statistic.eventLog.events.EventFields.StringListValidatedByCustomRule
+import com.intellij.internal.statistic.eventLog.validator.ValidationResultType.ACCEPTED
+import com.intellij.internal.statistic.eventLog.validator.rules.EventContext
+import com.intellij.internal.statistic.eventLog.validator.rules.impl.CustomValidationRule
+import com.intellij.internal.statistic.service.fus.collectors.CounterUsagesCollector
+import com.intellij.xdebugger.frame.XStackFrame
+
+private const val UNKNOWN_TYPE = "Unknown"
+
+class XDebuggerActionsCollector : CounterUsagesCollector() {
+  companion object {
+
+    private val group = EventLogGroup("xdebugger.actions", 3)
+    const val PLACE_THREADS_VIEW = "threadsView"
+    const val PLACE_FRAMES_VIEW = "framesView"
+    private const val LOCATION = "location"
+    @Suppress("MemberVisibilityCanBePrivate") // Exposed for consumers of this metric
+    const val EVENT_FRAMES_UPDATED = "frames.updated"
+    @Suppress("MemberVisibilityCanBePrivate") // Exposed for consumers of this metric
+    const val TOTAL_FRAMES = "totalFrames"
+    @Suppress("MemberVisibilityCanBePrivate") // Exposed for consumers of this metric
+    const val FILE_TYPES = "fileType"
+    @Suppress("MemberVisibilityCanBePrivate") // Exposed for consumers of this metric
+    const val FRAMES_PER_TYPE = "framesPerFileType"
```

Comment:
Android Studio has a StatisticsEventLogger implementation that intercepts these events and converts to our own metrics. Rather than using the field names as string constants, it seems better to make these values public and refer to them by name.

Reply:
I think we could discuss better way for this with some API, but maybe not is scope of this commit.